sled-storage/src/disk.rs

```diff
@@ -404,6 +426,26 @@ impl Disk {
             Self::Synthetic(disk) => disk.raw.slot,
         }
     }
+
+    // Today the only update we expect to be able to apply to
+    // a `Disk` is firmware.
+    pub(crate) fn update_disk(&mut self, raw_disk: &RawDisk) {
```

Comment:
I wonder if we could program this a little bit more defensively, in case there are other parts of these structures that can be updated. The comment indicates the true behavior ("we ignore everything but the firmware") but the calllsites don't really make that clear.

Can we either call this `update_firmware`, or have some other way to make sure "we don't drop updates on the floor" if something other than the firmware field changes?

Reply:
I can fix this - I had originally left this as `update_disk` as there might be more fields in the future we wish to support. But I am happy to make this an explicit `update_firmware` for the time being. (side note... maybe `update_firmware_metadata` is a better name).